Tests/ttx/ttx_test.py

```diff
@@ -263,5 +267,569 @@ def test_guessFileType_invalid_path(self):
+    expected_tables = ("head", "hhea", "maxp", "OS/2", "name", "cmap", "hmtx", "fpgm", "prep", "cvt ", "loca", "glyf", "post", "gasp", "DSIG")
+    for table in expected_tables:
+        assert table in ttf
+
+
+def test_ttx_ttcompile_ttf_to_woff_with_zopfli(tmpdir):
+    inttx = os.path.join("Tests", "ttx", "data", "TestTTF.ttx")
+    outwoff = os.path.join(str(tmpdir), "TestTTF.woff")
+    options = ttx.Options([("", "")], 1)
+    options.flavor = "woff"
+    options.useZopfli = True
+    ttx.ttCompile(inttx, outwoff, options)
+    # confirm that font was built
+    assert os.path.isfile(outwoff) is True
+    # confirm that it is valid TTF file, can instantiate a TTFont, has expected OpenType tables
+    ttf = TTFont(outwoff)
+    expected_tables = ("head", "hhea", "maxp", "OS/2", "name", "cmap", "hmtx", "fpgm", "prep", "cvt ", "loca", "glyf", "post", "gasp", "DSIG")
+    for table in expected_tables:
+        assert table in ttf
+
+
+def test_ttx_ttcompile_ttf_to_woff2(tmpdir):
+    inttx = os.path.join("Tests", "ttx", "data", "TestTTF.ttx")
+    outwoff2 = os.path.join(str(tmpdir), "TestTTF.woff2")
+    options = ttx.Options([("", "")], 1)
+    options.flavor = "woff2"
+    ttx.ttCompile(inttx, outwoff2, options)
+    # confirm that font was built
+    assert os.path.isfile(outwoff2) is True
+    # confirm that it is valid TTF file, can instantiate a TTFont, has expected OpenType tables
+    ttf = TTFont(outwoff2)
+    # DSIG should not be included from original ttx as per woff2 spec (https://dev.w3.org/webfonts/WOFF2/spec/)
+    assert "DSIG" not in ttf
+    expected_tables = ("head", "hhea", "maxp", "OS/2", "name", "cmap", "hmtx", "fpgm", "prep", "cvt ", "loca", "glyf", "post", "gasp")
+    for table in expected_tables:
+        assert table in ttf
+
+
+def test_ttx_ttcompile_ttf_timestamp_calcs(tmpdir):
+    inttx = os.path.join("Tests", "ttx", "data", "TestTTF.ttx")
+    outttf1 = os.path.join(str(tmpdir), "TestTTF1.ttf")
+    outttf2 = os.path.join(str(tmpdir), "TestTTF2.ttf")
+    options = ttx.Options([("", "")], 1)
+    # build with default options = do not recalculate timestamp
+    ttx.ttCompile(inttx, outttf1, options)
+    # confirm that font was built
+    assert os.path.isfile(outttf1) is True
+    # confirm that timestamp is same as modified time on ttx file
+    mtime = os.path.getmtime(inttx)
+    epochtime = timestampSinceEpoch(mtime)
+    ttf = TTFont(outttf1)
+    assert ttf['head'].modified == epochtime
+
+    # reset options to recalculate the timestamp and compile new font
+    options.recalcTimestamp = True
+    ttx.ttCompile(inttx, outttf2, options)
+    # confirm that font was built
+    assert os.path.isfile(outttf2) is True
+    # confirm that timestamp is more recent than modified time on ttx file
+    mtime = os.path.getmtime(inttx)
+    epochtime = timestampSinceEpoch(mtime)
+    ttf = TTFont(outttf2)
+    assert ttf['head'].modified > epochtime
+
+
+def test_ttx_ttcompile_otf_timestamp_calcs(tmpdir):
```

Comment:
this and the previous test only differ by the input ttx file, you can use `@pytest.mark.parametrize` to run the same test function with different inputs, see pytest docs